File: Java/src/com/dbc/algorithm/Math/UglyNumber.java
package com.dbc.algorithm.Math;

public class UglyNumber {
    public boolean isUgly(int n) {
        if (n <= 0) return false;
        int[] factor = new int[]{2, 3, 5};
        for (int num : factor) {
            while (n % num == 0) {
                n /= num;
            }
        }
        return n == 1;
    }
}

File: Python/math/ugly_number.py
def isUgly(n: int) -> bool:
    if n <= 0:
        return False

    factors = [2, 3, 5]
    for factor in factors:
        while n % factor == 0:
            n //= factor
    
    return n == 1

if __name__ == "__main__":
    print(isUgly(0))